waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -297,9 +299,15 @@ public PanopticOperationHandler getPanopticOperationHandler() {
           GetAllDatabasesRequest allDatabasesRequest = new GetAllDatabasesRequest(mapping, filter);
           allRequests.add(allDatabasesRequest);
         }
-        List<String> result = executeRequests(allRequests, GET_DATABASES_TIMEOUT, "Can't fetch databases: {}");
+        List<String> result = getPanopticOperationExecutor()
+            .executeRequests(allRequests, GET_DATABASES_TIMEOUT, "Can't fetch databases: {}");
         return result;
       }
+
+      @Override
+      protected PanopticOperationExecutor getPanopticOperationExecutor() {
+        return new PanopticConcurrentOperationExecutor();
```

Comment:
Is it OK/necessary to create a new one every time?

Reply:
it is OK it's just the object creation the actual method called is the same as before so just object creation overhead which is fine probably even preferred for every request etc. Keeps things simple